pandas/core/generic.py

```diff
@@ -8732,11 +8906,11 @@ def stat_func(self, axis=None, skipna=None, level=None, ddof=1,
 
 
 def _make_cum_function(cls, name, name1, name2, axis_descr, desc,
-                       accum_func, accum_func_name, mask_a, mask_b):
+                       accum_func, accum_func_name, mask_a, mask_b, examples):
     @Substitution(outname=name, desc=desc, name1=name1, name2=name2,
-                  axis_descr=axis_descr, accum_func_name=accum_func_name)
-    @Appender("Return {0} over requested axis.".format(desc) +
-              _cnum_doc)
+                  axis_descr=axis_descr, accum_func_name=accum_func_name,
+                  examples=examples)
+    @Appender(_cnum_doc)
```

Comment:
I think having separate string examples for each method makes everything clearer, especially when showing examples for use of `skipna` & `axis`. It also helps with keeping the docstring concise. For instance, now we can have a `Series` example for each method.

The disadvantage is user will only see examples for the method they’re checking, but I think this is ok because we are referencing all methods in the ‘See also’ section, which comes before 'Examples'.

In these PRs #20216 and #20217 examples for `DataFrame.all` and `DataFrame.any` are separate even though they are similar methods.

Reply:
Yes, I am also in favor of splitting up the examples.